core/src/main/java/org/elasticsearch/action/admin/indices/template/put/PutIndexTemplateRequest.java

```
@@ -311,6 +329,12 @@ public PutIndexTemplateRequest source(Map templateSource) {
                 }
             } else if (name.equals("aliases")) {
                 aliases((Map<String, Object>) entry.getValue());
+            } else if (name.equals("validate")) {
+                if (entry.getValue() instanceof Boolean) {
+                    validation((Boolean)entry.getValue());
+                } else if (entry.getValue() instanceof String) {
+                    validation(Booleans.parseBoolean((String)entry.getValue(), DEFAULT_VALIDATION));
+                }
```

can you throw an exception in a else block?